src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -36,14 +36,14 @@ class Serializer implements SerializerInterface
     private const STAMP_HEADER_PREFIX = 'X-Message-Stamp-';
 
     private SymfonySerializerInterface $serializer;
-    private string $format;
-    private array $context;
 
-    public function __construct(?SymfonySerializerInterface $serializer = null, string $format = 'json', array $context = [])
-    {
+    public function __construct(
+        ?SymfonySerializerInterface $serializer = null,
+        private string $format = 'json',
+        private array $context = [],
+    ) {
         $this->serializer = $serializer ?? self::create()->serializer;
-        $this->format = $format;
-        $this->context = $context + [self::MESSENGER_SERIALIZATION_CONTEXT => true];
+        $this->context[self::MESSENGER_SERIALIZATION_CONTEXT] = true;
```

Comment:
What's the difference? 🤔

Reply:
Am I wrong wrong am I missing something:

The suggestion would create a temporary associative array with the `MESSENGER_SERIALIZATION_CONTEXT` key  and then merge it with `$context`. 
Merging arrays in PHP *always* creates a copy of the orginial array and does not modify the original. Right?

Because arrays in PHP are not passed by reference, why would it be necessary to create a copy of the context when a copy of the array is already created when the object is constructed?

Reply:
`+=` doesn't override existing keys, that's the diff, and it's a significant one here

Reply:
TIL